### ISSUE TITLE
HookAssemblyReslove exception

### ISSUE THREAD

**Issue by appqui** [StackExchange/StackExchange.Precompilation#13]
```
30>  C:\Projects\metrica\branch\architect\packages\StackExchange.Precompilation.Build.3.3.0\build\..\tools\StackExchange.Precompiler.exe /noconfig /nowarn:1701,1702,2008 /nostdlib+ /errorreport:prompt /warn:4
....
30>CSC : error : An unhandled exception occured
30>  System.ArgumentException: An item with the same key has already been added.
30>     at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
30>     at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
30>     at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
30>     at StackExchange.Precompilation.CompilationProxy.HookAssemblyReslove(IEnumerable`1 references) in C:\TeamCity\buildAgent\work\967c65bed2f567c6\StackExchange.Precompilation.Build\CompilationProxy.cs:line 94
30>     at StackExchange.Precompilation.CompilationProxy.HookAssemblyReslove(IEnumerable`1 references)
30>     at StackExchange.Precompilation.CompilationProxy.RunCs(String[] args) in C:\TeamCity\buildAgent\work\967c65bed2f567c6\StackExchange.Precompilation.Build\CompilationProxy.cs:line 57
30>     at StackExchange.Precompilation.Program.Main(String[] args) in C:\TeamCity\buildAgent\work\967c65bed2f567c6\StackExchange.Precompilation.Build\Program.cs:line 15
```

I am getting this after installing from NuGet 3.3.0 version. What strange, I don't see such methods/line numbers in source code.

**Comment by m0sa** [comment]
you're probably looking at the new (roslyn 2.0 code that's not on nuget yet), the nuget is built from this tag - https://github.com/StackExchange/StackExchange.Precompilation/tree/releases/3.3.0

**Comment by MatthewSteeples** [comment]
We're seeing this two. Are there any blockers for releasing a 2.0 compatible version of this library on NuGet (or is there one on a MyGet somewhere we can use)?

**Comment by MatthewSteeples** [comment]
From what we can determine it's something to do with NETStandard. Mainly happens on our build server, and only when we update our Azure packages to the latest version. Still on 1.3.2 of Roslyn so it's not related to v2 of that

**Comment by m0sa** [comment]
It means you have references to two versions of the same assembly.

The offending line is here -> https://github.com/StackExchange/StackExchange.Precompilation/blob/releases/3.3.0/StackExchange.Precompilation.Build/CompilationProxy.cs#L94

This shouldn't happen in 4.0 as I've made some changes to this logic. I pushed a [pre-release 4.0 package to nuget](https://www.nuget.org/packages/StackExchange.Precompilation.Build/4.0.0-roslynv2beta), can anybody update the packages and report what happens?

**Comment by DaniilSokolyuk** [comment]
@m0sa hello, why 4.0 target net462 framework?

**Comment by m0sa** [comment]
Because roslyn v2 / Microsoft.CodeAnalysis.CSharp targets it.

**Comment by DaniilSokolyuk** [comment]
@m0sa 

Okay, thank you.
Local works fine, but on teamcity build i got exception.
```
D:\TeamCity2\buildAgent\work\57f3d64e0aa30db6\packages\StackExchange.Precompilation.Build.4.0.0-roslynv2beta\build\..\tools\StackExchange.Precompiler.exe ....
[17:09:39][Step 2/6]   System.IO.FileNotFoundException: Could not load file or assembly 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
[17:09:39][Step 2/6]   File name: 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```

**Comment by MatthewSteeples** [comment]
I've just tried the 4.0 build, and I'm getting a different exception entirely and can't track it down. For most of the .cs files in my web project I get an output line with the following message
`Failed parasing source tree: System.TypeInitializationException: The type initializer for 'StackExchange.Precompilation.CSharpParser' threw an exception. ---> System.ArgumentNullException: Value cannot be null.`

I can't find a reference to CSharpParser anywhere at all in the current branch, and it looks like it was removed [two weeks ago](https://github.com/StackExchange/StackExchange.Precompilation/commit/7865d8e8eb8fb1f587af30015b765d5e34d48eff). What source was the 4.0 prerelease built from?
